v2/pkg/api/v1alpha2/types_config.go

```diff
@@ -159,19 +160,27 @@ func (o Operator) GetUniqueName() (string, error) {
 	}
 
 	if o.TargetTag != "" {
-		ctlgSpec.Digest = ""
 		ctlgSpec.Reference = strings.Replace(ctlgSpec.Reference, ctlgSpec.Tag, o.TargetTag, 1)
 		ctlgSpec.ReferenceWithTransport = strings.Replace(ctlgSpec.ReferenceWithTransport, ctlgSpec.Tag, o.TargetTag, 1)
 		ctlgSpec.Tag = o.TargetTag
 	}
 	if o.TargetCatalog != "" {
-		ctlgSpec.Reference = strings.Replace(ctlgSpec.Reference, ctlgSpec.PathComponent, o.TargetCatalog, 1)
-		ctlgSpec.ReferenceWithTransport = strings.Replace(ctlgSpec.ReferenceWithTransport, ctlgSpec.PathComponent, o.TargetCatalog, 1)
-		ctlgSpec.PathComponent = o.TargetCatalog
+		if IsValidPathComponent(o.TargetCatalog) {
+			ctlgSpec.Reference = strings.Replace(ctlgSpec.Reference, ctlgSpec.PathComponent, o.TargetCatalog, 1)
+			ctlgSpec.ReferenceWithTransport = strings.Replace(ctlgSpec.ReferenceWithTransport, ctlgSpec.PathComponent, o.TargetCatalog, 1)
+			ctlgSpec.PathComponent = o.TargetCatalog
+		} else {
+			return "", fmt.Errorf("targetCatalog: %s - value is not valid. It should not contain a tag or a digest. It is expected to be composed of 1 or more path components separated by /, where each path component is a set of alpha-numeric and  regexp (?:[._]|__|[-]*). For more, see https://github.com/containers/image/blob/main/docker/reference/regexp.go", o.TargetCatalog)
```

Comment:
It is good to have this level of explanation here with the source reference. Thanks.